src/irmin-pack/IO.ml

```diff
@@ -28,15 +28,25 @@ module Unix : S = struct
     file : string;
     raw : Raw.t;
     mutable offset : int63;
+    (** [offset] is the virtual offset of the end of the file. If we {!append} data, it is
+        written at [offset]. Note that data first goes into [buf] before being flushed to
+        file. *)
+
     mutable flushed : int63;
+    (** [flushed] is the "maximum last flushed offset"; it is updated in {!unsafe_flush}
+        and {!truncate} *)
+
     readonly : bool;
     mutable version : Version.t;
     buf : Buffer.t;
+    (** When data is written to a file, it is first held in a buffer, which is flushed to
+        disk when the amount of data exceeds [auto_flush_limit]. *)
   }
 
   let name t = t.file
   let header_size = (* offset + version *) Int63.of_int 16
 
+  (* FIXME what is unsafe? *)
```

Comment:
But what is "unsafe"?

Reply:
It's unsafe in the sense that `unsafe_flush readonly_instance` will silently return `()` rather than raising the appropriate `RO_not_allowed` exception.